cannot read random object id: open /dev/urandom: permission denied

What are the steps to reproduce this issue?
-------------------------------------------
Testing  CRI in interactive mode  in sections 3 and 4:
    1. …in window(bash) A start sycri:  sycri -v=10
    2. …in window B (bash) cd singularity-cri/examples; crictl runp net-prod.json

What happens?
-------------
In window A the following error is thrown: 
 panic: cannot read random object id: open /dev/urandom: permission denied
goroutine 1 [running]:
github.com/sylabs/singularity/vendor/github.com/globalsign/mgo/bson.readRandomUint32(0xa0ed04)
        /home/rsmith/go/src/github.com/sylabs/singularity/vendor/github.com/globalsign/mgo/bson/bson.go:236 +0x100
FATAL:   exit status 1
I0412 12:23:51.660863   25995 pod_runtime.go:130] Forcibly stopping pod a069a59e8ede685ebd558dcd13d44547f8362dd5fb7b1d762462556a4a9107ee
I0412 12:23:51.660916   25995 client.go:66] Executing [singularity -q oci kill -s SIGKILL a069a59e8ede685ebd558dcd13d44547f8362dd5fb7b1d762462556a4a9107ee]
...

What were you expecting to happen?
----------------------------------
See output like:
0e0538d57a52d8673b9ad5124dd017087b3f5292f82cb9406c10f270f8f531fa

And be able to execute the following command:
$ sudo crictl pods
POD ID              CREATED             STATE               NAME                NAMESPACE           ATTEMPT
0e0538d57a52d       26 seconds ago      Ready               networking          default             1

Any logs, error output, comments, etc?
----------------------------
(If it’s long, please paste a link to the full output here.)
...

Environment?
----------------------------------------
**OS distribution and version:** 
CentOS Linux release 7.4.1708 (Core)  3.10.0-327.el7.x86_64 #1 SMP Thu Nov 19 22:10:57 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
**go version:** …
go version go1.11.5 linux/amd64
**go env:** …
GOPATH=/home/rsmith/go
**Singularity-CRI version:** …
current
**Singularity version:**
3.1.1-rc1
**Kubernetes version:** 
kubernetes v1.12.5 

Randy Smith
Sr. HPC Engineer
Translational Genomics Research Institute

root@sidewinder-login rsmith]# autrace /usr/local/bin/sycri -v=10
Waiting to execute: /usr/local/bin/sycri
I0425 13:46:45.883527   20884 network.go:112] Network configuration found: weave
I0425 13:46:45.883764   20884 main.go:175] Singularity CRI server started on /var/run/singularity.sock
I0425 13:46:45.899460   20884 device.go:97] Loading NVML
E0425 13:46:45.899560   20884 device.go:99] Could not initialize NVML library: could not load NVML library
W0425 13:46:45.899589   20884 main.go:191] GPU support is not enabled: unable to load: check libnvidia-ml.so.1 library and graphic drivers
I0425 13:46:49.187514   20884 pod_validate.go:71] Setting pod cgroup parent to default value "singularity-cri/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae"
I0425 13:46:49.187555   20884 pod_files.go:88] Creating /var/run/singularity/pods/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae/namespaces
I0425 13:46:49.187601   20884 pod_files.go:123] Creating log directory /tmp/net-pod
I0425 13:46:49.187713   20884 pod_files.go:106] Creating hostname file /var/run/singularity/pods/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae/hostname
I0425 13:46:49.199403   20884 pod_files.go:132] Creating /var/run/singularity/pods/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae/bundle/rootfs
I0425 13:46:49.199477   20884 pod_files.go:141] Creating oci config /var/run/singularity/pods/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae/bundle/config.json
I0425 13:46:49.200321   20884 pod_runtime.go:49] Creating pod cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae
I0425 13:46:49.200361   20884 client_oci.go:121] Executing [singularity -q oci create --empty-process --sync-socket /var/run/singularity/pods/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae/sync.sock -b /var/run/singularity/pods/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae/bundle cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae]
I0425 13:46:49.259289   20884 sync.go:75] Received state 1 at /var/run/singularity/pods/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae/sync.sock
panic: cannot read random object id: open /dev/urandom: permission denied

goroutine 1 [running]:
github.com/sylabs/singularity/vendor/github.com/globalsign/mgo/bson.readRandomUint32(0xa0ed04)
	/home/rsmith/go/src/github.com/sylabs/singularity/vendor/github.com/globalsign/mgo/bson/bson.go:236 +0x100
FATAL:   exit status 1
I0425 13:46:49.301875   20884 pod_runtime.go:130] Forcibly stopping pod cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae
I0425 13:46:49.301964   20884 client.go:66] Executing [singularity -q oci kill -s SIGKILL cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae]
FATAL:   cannot kill 'cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae', the state of the container must be created or running
E0425 13:46:49.319736   20884 pod.go:100] Could not kill pod after failed run: could not terminate pod: could not execute: exit status 255
I0425 13:46:49.319773   20884 sync.go:63] Context is done
E0425 13:46:49.319835   20884 sync.go:113] Accept failed: accept unix /var/run/singularity/pods/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae/sync.sock: use of closed network connection
E0425 13:46:49.334765   20884 pod.go:103] Could not remove pod: could not delete instance cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae: FATAL:   no container found with name cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae
I0425 13:46:49.334807   20884 pod_files.go:158] Removing binded namespace /var/run/singularity/pods/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae/namespaces/uts
I0425 13:46:49.334888   20884 pod_files.go:158] Removing binded namespace /var/run/singularity/pods/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae/namespaces/network
I0425 13:46:49.334946   20884 pod_files.go:158] Removing binded namespace /var/run/singularity/pods/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae/namespaces/ipc
I0425 13:46:49.335012   20884 pod_files.go:158] Removing binded namespace 
I0425 13:46:49.335029   20884 pod_files.go:164] Removing pod base directory /var/run/singularity/pods/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae
I0425 13:46:49.335718   20884 pod_files.go:169] Removing pod log directory /tmp/net-pod
E0425 13:46:49.336685   20884 main.go:54] /runtime.v1alpha2.RuntimeService/RunPodSandbox
	Request: {"config":{"metadata":{"name":"networking","uid":"networking-pod-uid","namespace":"default","attempt":1},"hostname":"networking","log_directory":"/tmp/net-pod","port_mappings":[{"container_port":80}],"linux":{"cgroup_parent":"singularity-cri/cad4ebd8ca6e0e90f6a8b03e3d4f02b8d70d06a5403209e6f249b81377e58fae"}}}
	Response: null
	Error: rpc error: code = Internal desc = could not run pod: could not spawn pod: could not create pod: could not execute create container command: exit status 255

--- a/pkg/kube/pod_oci.go
+++ b/pkg/kube/pod_oci.go
@@ -55,6 +55,12 @@ func (t *podTranslator) translate() (*specs.Spec, error) {
 		Source:      "proc",
 		Type:        "proc",
 	})
+	t.g.AddMount(specs.Mount{
+		Destination: "/dev",
+		Type:        "tmpfs",
+		Source:      "tmpfs",
+		Options:     []string{"nosuid", "strictatime", "mode=755", "size=65536k"},
+	})
 	t.g.SetProcessCwd("/")
 	t.g.SetProcessArgs([]string{"true"})
 

